### PR TITLE
The python example now works in both Python 2.7 and 3.6

### PR DESCRIPTION
**Pull request by jkaster**
The `six` compatibility library is required for the new code

See https://six.readthedocs.io/ for more information on `six`

### CODE REVIEW

File: python_example.py [hunk 1]
```diff
@@ -32,33 +44,32 @@ class URL:
...
+                                self.nonce,
+                                self.time,
+                                self.session_length,
+                                self.user.external_user_id,
+                                self.user.permissions,
+                                self.user.models,
+                                self.user.group_ids,
+                                self.user.external_group_id,
+                                self.user.user_attributes,
+                                self.user.access_filters])
```
Comment (jbandhauer) [inline python_example.py]: I trust this works the same as the old code if any of these are empty string? And, any type conversions to string still work the same way? The *seems* like a nice refactoring. But, as a non-python-expert is not not 100% obvious that it is functionally identical to the previous code.

Comment (jkaster) [reply python_example.py]: @jbandhauer sounds like I should further modify the python SSO example to omit parameters that are "empty" from being encoded in the argument, like you do with Ruby. I'll do that Monday. Thanks for the suggestion. I'll add you to the review for that PR, even though it looks like you'll catch it anyway ;)

Comment (rlgomes) [reply python_example.py]: you could write this as:
```python
string_to_sign = "\n".join(filter(None, [self.looker.host,
                                         self.path,
                                         self.nonce,
                                         self.time,
                                         self.session_length,
                                         self.user.external_user_id,
                                         self.user.permissions,
                                         self.user.models,
                                         self.user.group_ids,
                                         self.user.external_group_id,
                                         self.user.user_attributes,
                                         self.user.access_filters]))
```

**Comment by jbandhauer** [comment]
> omit parameters that are "empty" from being encoded

The subtlety is that any args that are going to actually be present in the URL - even if they are empty arrays or empty strings - need to be represented in this signed portion. Where 'represented' might mean that there is only a `\n` in the that 'slot' of the pre-signed string.

I kinda regret even brining this up. The 'optional' params are only optional because looker versions before when the params were added didn't know about those params. In practice, those old versions are likely long gone. So, this is probably a non-issue at present. But, we are bound to add more things like that at some future point. So, we could either make the sample code flexibly on that point (like the Ruby code is - where is checks for the the param being 'present' rather than for it being 'blank'), or we could just say that people should use older versions of the sample code for compatibility with older versions of looker and not add this complexity to the sample code at all.

**Pull request merged**